### PR TITLE
polygon: tidy up astrid interface usage

### PR DESCRIPTION
**Pull request by taratorio**
closes https://github.com/erigontech/erigon/issues/11031, https://github.com/erigontech/erigon/issues/11030

final tidy ups for Astrid pre-launch
summary of changes:
- we were over-exporting/over-defining interfaces quite a lot - there a lot of places where we don't really need an interface
- instead stick to Go's recommendations:
   - always export structs via `New` or factory/creator methods in packages unless there is a reason to export an interface (e.g. valid reasons to introduce polymorphism such as multiple implementations)
   - defining slim interfaces at consumer packages  instead of the producer package to specify and decouple dependencies
- used Options pattern at several places for cleaner init code
- random cosmetic changes along the way
- no logical changes

### CODE REVIEW

File: polygon/bridge/reader.go [hunk 1]
```diff
@@ -149,7 +165,6 @@ func (r *RemoteReader) EventTxnLookup(ctx context.Context, borTxHash libcommon.H
 	return reply.BlockNumber, reply.Present, nil
 }
 
-// Close implements bridge.ReaderService. It's a noop as there is no attached store.
```
Comment (shohamc1) [inline polygon/bridge/reader.go:152]: Why is this removed?

File: polygon/sync/sync.go [hunk 2]
```diff
@@ -123,23 +123,19 @@ func (s *Sync) commitExecution(ctx context.Context, newTip *types.Header, finali
 	return nil
 }
 
-func (s *Sync) handleMilestoneTipMismatch(
-	ctx context.Context,
-	ccBuilder CanonicalChainBuilder,
-	milestone EventNewMilestone,
-) error {
+func (s *Sync) handleMilestoneTipMismatch(ctx context.Context, ccb *CanonicalChainBuilder, ms EventNewMilestone) error {
```
Comment (shohamc1) [inline polygon/sync/sync.go:126]: The old names were more descriptive.

**Review by shohamc1 — APPROVED** [review]

File: polygon/bridge/reader.go (hunk 1 above)
Comment (taratorio) [reply polygon/bridge/reader.go:152]: I thought it is an unnecessary comment after the bridge.ReaderService interface got removed. If someone removes this function they will get a compilation error in the consuming packages that expect to see this Close function on a bridge reader. However I'll add back the "no-op as there is no attached store" comment bit back as I agree it is useful to have.

File: polygon/sync/sync.go (hunk 2 above)
Comment (taratorio) [reply polygon/sync/sync.go:126]: The function names got very long and spread on many lines unnecessary for just 3 input args so wanted to make them a bit more concise. I think `ccb` is a nice abbreviation (like `ufc` for update fork choice) once you understand the context.  Agree `ms` is not so clear, so I renamed that to `event` for consistency with the other functions
